src/visual.ts

```diff
@@ -837,13 +871,63 @@ module powerbi.extensibility.visual {
             };
         }
 
-        private updateWidthOfColumn(): void {
+        private updateWidthOfColumn(updateMode: UpdateColumnsWidthMode): void {
             let countOfValues: number = this.dataView.dataPoints.length,
-                widthOfColumn: number;
-
-            widthOfColumn = countOfValues
-                ? this.viewportIn.width / countOfValues - Histogram.ColumnPadding
-                : Histogram.MinViewportInSize;
+                widthOfColumn: number,
+                columnsRect: any,
+                xAxisTicks: any,
+                leftBorderOffset,
+                rightBorderOffset,
+                borderValues: HistogramBorderValues = this.dataView.borderValues,
+                minX: string,
+                maxX: string,
+                minXOffset: number,
+                maxXOffset: number;
+
+            switch (updateMode) {
+                case UpdateColumnsWidthMode.markOnly :
+                    widthOfColumn = Histogram.PointMarkSize;
+                break;
+                case UpdateColumnsWidthMode.standardCalculation :
+                    widthOfColumn = countOfValues ? (this.viewportIn.width / countOfValues - Histogram.ColumnPadding) : Histogram.MinViewportInSize;
+                break;
+                case UpdateColumnsWidthMode.calculatePointsDiff :
+                default :
+                    // If number of points (bins) is 1, it means that we need to calculate difference between borders (else case)
```

Comment:
Move this case to separate function, e.g:
```
default:
  widthOfColumn = this.calculateDefaultWidth(); 
  break;
```

Reply:
I will do